### PR TITLE
Debt simplification algorithm

### PR DESCRIPTION
**Pull request by podocarp**
Please try it out!
Reference #70

### CODE REVIEW

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 1]
```diff
@@ -75,6 +91,7 @@ public void invite(Person ... people) {
      */
     public void disinvite(Person ... people) {
         // haven't implemented what if list does not contain that specific person
+        // TODO: also care about transfermatrix
```
Comment (daekoon) [inline src/main/java/seedu/address/model/activity/Activity.java]: Ideally this should be done in disinvite command. Validations should be done in command class.

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 2]
```diff
@@ -124,8 +182,71 @@ public void addExpense(Expense ... expenditures) throws PersonNotInActivityExcep
...
+            double acc = 0;
+            for (int b = 0; b < debtMatrix.size(); b++) {
+                acc += debtMatrix.get(a).get(b);
+                acc -= debtMatrix.get(b).get(a);
+                transferMatrix.get(a).set(b, 0.0);
+            }
+            participantBalances.set(a, acc);
+        }
+
+        System.out.println("Balances: " + participantBalances.toString());
```
Comment (Aulud) [inline src/main/java/seedu/address/model/activity/Activity.java]: Is this `println` statement intended as logging output?

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 3]
```diff
@@ -87,6 +109,48 @@ public Title getTitle() {
         return title;
     }
 
+    /**
+     * Gets the transfer matrix. Every (i, j) entry reflects how much i receives
+     * from j. Negative amounts means i has to give j money.
+     */
+    public ArrayList<ArrayList<Double>> getTransferMatrix() {
```
Comment (Aulud) [inline src/main/java/seedu/address/model/activity/Activity.java:118]: @podocarp I assume we call this method when we need to obtain the transfer matrix to show to the user?

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 4]
```diff
@@ -96,20 +160,14 @@ public boolean hasPerson(Integer personId) {
...
     /**
      * Remove people from the activity
-     * @param personId The people that will be removed from the activity.
+     * @param people The people that will be removed from the activity.
      */
-    public void disinvite(Integer personId) {
-        participantIds.remove(personId);
+    public void disinvite(Person ... people) {
+        // haven't implemented what if list does not contain that specific person
+        // TODO: also care about transfermatrix and debtmatrix
```
Comment (Aulud) [inline src/main/java/seedu/address/model/activity/Activity.java:171]: Let's leave this for @JohnNzj to complete when he sets up the disinvite command.

**Review by Aulud — DISMISSED** [review]
Seems okay overall - light testing doesn't reveal any unintended behaviour so far.

File: src/main/java/seedu/address/model/activity/Activity.java [hunk 5]
```diff
@@ -87,6 +109,48 @@ public Title getTitle() {
...
+    public ArrayList<ArrayList<Double>> getTransferMatrix() {
+        simplifyExpenses();
+        return transferMatrix;
+    }
+
+    /**
+     * Invite people to the activity.
+     * @param people The people that will be added into the activity.
+     */
+    public void invite(Person ... people) {
```
Comment (daekoon) [inline src/main/java/seedu/address/model/activity/Activity.java:127]: Is this invite method being used somewhere? Also, is there a reason on why this endpoint isn't updating debtMatrix/transferMatrix?

File: src/main/java/seedu/address/model/activity/Amount.java [hunk 6]
```diff
@@ -22,6 +22,10 @@ public Amount(double amount) {
         value = amount;
     }
 
+    public double getValue() {
```
Comment (daekoon) [inline src/main/java/seedu/address/model/activity/Amount.java]: Value field is already public. Is this necessary :thinking:

File: src/main/java/seedu/address/model/activity/Activity.java (hunk 3 above)
Comment (podocarp) [reply src/main/java/seedu/address/model/activity/Activity.java:118]: yes

File: src/main/java/seedu/address/model/activity/Activity.java (hunk 5 above)
Comment (podocarp) [reply src/main/java/seedu/address/model/activity/Activity.java:127]: it does, it uses the invite below this one

File: src/main/java/seedu/address/model/activity/Activity.java (hunk 2 above)
Comment (podocarp) [reply src/main/java/seedu/address/model/activity/Activity.java]: gonna remove that

File: src/main/java/seedu/address/model/activity/Amount.java (hunk 6 above)
Comment (podocarp) [reply src/main/java/seedu/address/model/activity/Amount.java]: no clue

**Comment by podocarp** [comment]
Recommended to check out #110 after this to minimize conflicts later on, but just a suggestion.

**Review by daekoon — APPROVED** [review]
LGTM

**Pull request merged**